frontend/src/scenes/plugins/edit/PluginJobConfiguration.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react'
+import { PlayCircleOutlined, CheckOutlined, CloseOutlined } from '@ant-design/icons'
+import { Tooltip, Form, Input, Radio } from 'antd'
+import Modal from 'antd/lib/modal/Modal'
+import MonacoEditor from '@monaco-editor/react'
+import api from 'lib/api'
+import { JobSpec } from '~/types'
+import { toast } from 'react-toastify'
+import { errorToast, validateJsonFormItem } from 'lib/utils'
+
+const noop = (): void => {}
+
+interface PluginJobConfigurationProps {
+    jobName: string
+    jobSpec: JobSpec
+    pluginConfigId: number
+}
+
+const requiredRule = {
+    required: true,
+    message: 'Please enter a value!',
+}
+
+export function PluginJobConfiguration({ jobName, jobSpec, pluginConfigId }: PluginJobConfigurationProps): JSX.Element {
+    const [isJobModalOpen, setIsJobModalOpen] = useState(false)
```

Comment:
Wanted to stay away from `pluginLogic` as it is already pretty crowded but happy to move this there or a separate logic file if preferred